### ISSUE TITLE
SVG Loader: incorrect rendering in complex SVGs

### ISSUE THREAD

**Issue by dciug** [mrdoob/three.js#21330]
**Describe the bug**

I'm trying to use SVGLoader to load a more complex SVG generated from Illustrator

I get hundreds of warnings:
 - THREE.Color: Unknown color url(#linear-gradient-26)
 - SVGLoader: url access in attributes is not implemented.
 - SVGLoader: Elliptic arc or ellipse rotation or skewing is not implemented.

And a few errors:
 - THREE.BufferGeometry.computeBoundingSphere(): Computed radius is NaN. The "position" attribute is likely to have NaN values. 

**To Reproduce**

It occurs when I'm trying to load a complex svg, the code is the same. Unfortunately, I cannot share this SVG. I'll just upload parts of it, I suppose.

This is how the browser renders it:
![image](https://user-images.githubusercontent.com/6199480/108632588-76070080-7478-11eb-90e7-6c123a49c056.png)

This is how Three.js renders it:
![image](https://user-images.githubusercontent.com/6199480/108632633-9f279100-7478-11eb-8847-4aebca3e0327.png)

**Expected behavior**

It should look exactly the same as the browser would render it natively, but the colors are undefined and the z-order of the elements seems wrong. The wireframe visualization is correct, though.
The SVG also contains subtle elements with small opacity values. SVGLoader seems to ignore the shapes that are supposed to be visible through the transparent layers.

**Platform:**

 - Device: Desktop
 - OS: Windows
 - Browser: Chrome, Firefox, Edge
 - Three.js version: main branch

**Comment by yomboprime** [comment]
Sorry, I don't have anything useful to apport at this moment, and can't reproduce without the svg file.

**Comment by dciug** [comment]
I suppose most of the problems arise from the warnings, as Three.js cannot link the <linearGradient> defs using the url.

**Comment by yomboprime** [comment]
Yes, there are lots of SVG features not implemented in the parser, like gradients, correct painting order, urls...

**Comment by yomboprime** [comment]
But, now that I think of it, you only want an image of the SVG, right? For that use case you can just convert the SVG to canvas and load into a texture (I think there's an example of that in the repo)

**Comment by dciug** [comment]
The first version used SVG interactivity using Snap.svg to click on the elements. Each click would get the coordinates of the element and show some contextual element at that point. However, the performance was quite poor and unstable, which is why I looked into Three.js. I could convert it into canvas, but I would still need to get the coordinates of the elements. Can I do that in Canvas?

**Comment by mrdoob** [comment]
Closing this. We can reopen when the svg file is provided.

**Comment by dciug** [comment]
Here's a stripped down version of the svg. Rename "txt" to "svg".

[three.txt](https://github.com/mrdoob/three.js/files/6023156/three.txt)

**Comment by yomboprime** [comment]
I've spent some time on this. It seems some bugs have been solved with previous PRs, and the errors about NaN values in the console are gone.

The mentioned warnings are still shown. These are the features that need to be implemented in SVGLoader to address them:

- Textures
- URL access in attributes
- URL-encoded PNG textures
- Fill gradients
- Elliptic arc or ellipse rotation or skewing
- Paint order

**Comment by iambarnaby** [comment]
This still doesn't seem to work. Unless I'm missing something.

**Comment by LeviPesin** [comment]
Because not all points of https://github.com/mrdoob/three.js/issues/21330#issuecomment-806367989 are solved yet.
